src/base/emu-i386/simx86/codegen-x86.c

```diff
@@ -3266,17 +3266,24 @@ unsigned int Exec_x86(TNode *G, int ln)
 	/* signal_pending at this point is 1 if there was ANY signal,
 	 * not just a SIGALRM
 	 */
-	if (signal_pending()) {
-		CEmuStat|=CeS_SIGPEND;
-	}
-	if (G->flags & F_INHI)
+	if ((G->flags & F_INHI) && !(G->seqnum == 1 && (CEmuStat & CeS_INHI))) {
+		/* ignore signals and traps for movss/popss; if there is just
```

Comment:
Would you like to not ignore signals?
We could instead just add an additional
check in pic.c where it currently checks
for IF flag. We can check there also the
INHI flag, its really trivial.
Not that I am insisting, I think that makes
sense, but up to you to decide.

Reply:
I don't really like exporting CEmuStat to pic.c... If I can keep it local to cpuemu I'd be happier :)

Reply:
An extra func will help then. :)
Well, its up to you, as an implementer,
but my considerations are:
1. Inhibiting signals is quite bad.
It means that the user code can
delay signal delivery. With that
inhi-drop stuff at least you can't
defer it indefinitely (with the stream
of MOVSS), but it was possible
before.
2. This is actually what the VXT
hypervisors do, i.e. checking IF
is not enough: you need to check
all inhibitions.